community/cypher/cypher-compiler-2.3/src/main/scala/org/neo4j/cypher/internal/compiler/v2_3/executionplan/CompiledExecutionResult.scala

```diff
@@ -189,12 +189,14 @@ abstract class CompiledExecutionResult(taskCloser: TaskCloser, statement:Stateme
   }.mkString("{", ", ", "}")
 
   private def doInAccept[T](body: ResultRow => T) = {
-    accept(new ResultVisitor[RuntimeException] {
-      override def visit(row: ResultRow): Boolean = {
-        body(row)
-        true
-      }
-    })
+    if (!taskCloser.isClosed) {
```

Comment:
I think we still should throw an exception here, maybe something nicer than the `Statement has been closed` but I still think it makes sense to throw here.